src/modules/getPageData.ts

```diff
@@ -19,6 +20,7 @@ export const getPageData = function (this: Swup, request: XMLHttpRequest): PageD
 
 	return {
 		...pageHtmlData,
-		responseURL: request.responseURL || window.location.href
+		responseURL: request.responseURL || window.location.href,
+		timestamp: Date.now()
```

Comment:
I think we should provide several timestamps actually, `createdAt`, `lastUpdated` and maybe `lastAccessed`. This way custom logic can be based on whatever suits your use case.